### ISSUE TITLE
Uncaught TypeError: Cannot read property 'indexOf' of undefined

### ISSUE THREAD

**Issue by qingfeng** [atom/find-and-replace#499]
[Enter steps to reproduce below:]

1. ...
2. ...

**Atom Version**: 1.0.7
**System**: Mac OS X 10.10.3
**Thrown From**: [find-and-replace](https://github.com/atom/find-and-replace) package, v0.180.0

### Stack Trace

Uncaught TypeError: Cannot read property 'indexOf' of undefined

```
At /Applications/Atom.app/Contents/Resources/app.asar/node_modules/find-and-replace/lib/find-view.js:566

TypeError: Cannot read property 'indexOf' of undefined
  at FindView.module.exports.FindView.updateResultCounter (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/find-and-replace/lib/find-view.js:566:66)
  at /Applications/Atom.app/Contents/Resources/app.asar/node_modules/find-and-replace/lib/find-view.js:3:61
  at Emitter.module.exports.Emitter.emit (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/emitter.js:82:11)
  at BufferSearch.module.exports.BufferSearch.setCurrentMarkerFromSelection (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/find-and-replace/lib/buffer-search.js:268:27)
  at Emitter.module.exports.Emitter.emit (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/emitter.js:82:11)
  at TextEditor.module.exports.TextEditor.selectionRangeChanged (/Applications/Atom.app/Contents/Resources/app.asar/src/text-editor.js:2112:27)
  at Selection.module.exports.Selection.screenRangeChanged (/Applications/Atom.app/Contents/Resources/app.asar/src/selection.js:911:26)
  at /Applications/Atom.app/Contents/Resources/app.asar/src/selection.js:43:24
  at Emitter.module.exports.Emitter.emit (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/emitter.js:82:11)
  at Marker.module.exports.Marker.notifyObservers (/Applications/Atom.app/Contents/Resources/app.asar/src/marker.js:270:27)
  at /Applications/Atom.app/Contents/Resources/app.asar/src/marker.js:65:26
  at Emitter.module.exports.Emitter.emit (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/emitter.js:82:11)
  at Marker.module.exports.Marker.emitChangeEvent (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/text-buffer/lib/marker.js:401:20)
  at Marker.module.exports.Marker.update (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/text-buffer/lib/marker.js:346:12)
  at Marker.module.exports.Marker.setRange (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/text-buffer/lib/marker.js:108:19)
  at Marker.module.exports.Marker.setBufferRange (/Applications/Atom.app/Contents/Resources/app.asar/src/marker.js:137:32)
  at /Applications/Atom.app/Contents/Resources/app.asar/src/selection.js:117:24
  at Selection.module.exports.Selection.modifySelection (/Applications/Atom.app/Contents/Resources/app.asar/src/selection.js:940:7)
  at Selection.module.exports.Selection.setBufferRange (/Applications/Atom.app/Contents/Resources/app.asar/src/selection.js:110:19)
  at Selection.module.exports.Selection.selectWord (/Applications/Atom.app/Contents/Resources/app.asar/src/selection.js:404:12)
  at TextEditorComponent.module.exports.TextEditorComponent.onMouseDown (/Applications/Atom.app/Contents/Resources/app.asar/src/text-editor-component.js:556:42)
  at HTMLDivElement.<anonymous> (/Applications/Atom.app/Contents/Resources/app.asar/src/text-editor-component.js:3:61)

```

### Commands

```
     -0:05.9.0 find-and-replace:show (atom-text-editor.editor.is-focused)
     -0:05.6.0 core:paste (atom-text-editor.editor.mini.is-focused)
  4x -0:04.9.0 core:confirm (atom-text-editor.editor.mini.is-focused)
     -0:03.8.0 editor:consolidate-selections (atom-text-editor.editor.mini.is-focused)
     -0:03.7.0 core:cancel (atom-text-editor.editor.mini.is-focused)
  3x -0:02.4.0 core:copy (atom-text-editor.editor.is-focused)
```

### Config

```json
{
  "core": {
    "ignoredNames": [
      ".git",
      ".svn",
      ".DS_Store"
    ],
    "themes": [
      "atom-dark-ui",
      "monokai"
    ],
    "disabledPackages": [
      "pep8",
      "task-list"
    ]
  }
}
```

### Installed Packages

```coffee
# User
bottom-dock, v0.2.2
color-picker, v2.0.11
counter, v0.3.3
django-templates, v0.5.0
editor-background, v1.0.1
editor-stats, v0.17.0
editorconfig, v1.0.1
file-icons, v1.6.3
language-asciidoc, v0.11.1
language-django, v0.1.1
language-erlang, v3.0.0
language-slim, v0.5.0
linter, v1.3.4
linter-flake8, v1.6.1
linter-jscs, v2.0.4
local-history, v3.2.3
monokai, v0.18.0
pigments, v0.10.0
react, v0.12.6
todo-manager, v0.1.6
travis-ci-status, v0.16.0
zentabs, v0.8.6

# Dev
No dev packages
```

**Comment by berwyn** [comment]
1. Double-click to select text
2. Ctrl+F
3. Press [Enter] to move through results
4. Editor hangs, error dialogue spawns with the following error report:

**Atom Version**: 1.0.7
**System**: Microsoft Windows 8.1 Pro
**Thrown From**: [find-and-replace](https://github.com/atom/find-and-replace) package, v0.180.0


### Stack Trace

Uncaught TypeError: Cannot read property 'indexOf' of undefined

```
At C:\Users\jgreeley\AppData\Local\atom\app-1.0.7\resources\app.asar\node_modules\find-and-replace\lib\find-view.js:566

TypeError: Cannot read property 'indexOf' of undefined
  at FindView.module.exports.FindView.updateResultCounter (C:\Users\jgreeley\AppData\Local\atom\app-1.0.7\resources\app.asar\node_modules\find-and-replace\lib\find-view.js:566:66)
  at C:\Users\jgreeley\AppData\Local\atom\app-1.0.7\resources\app.asar\node_modules\find-and-replace\lib\find-view.js:3:61
  at Emitter.module.exports.Emitter.emit (C:\Users\jgreeley\AppData\Local\atom\app-1.0.7\resources\app.asar\node_modules\event-kit\lib\emitter.js:82:11)
  at BufferSearch.module.exports.BufferSearch.setCurrentMarkerFromSelection (C:\Users\jgreeley\AppData\Local\atom\app-1.0.7\resources\app.asar\node_modules\find-and-replace\lib\buffer-search.js:268:27)
  at Emitter.module.exports.Emitter.emit (C:\Users\jgreeley\AppData\Local\atom\app-1.0.7\resources\app.asar\node_modules\event-kit\lib\emitter.js:82:11)
  at TextEditor.module.exports.TextEditor.selectionRangeChanged (C:\Users\jgreeley\AppData\Local\atom\app-1.0.7\resources\app.asar\src\text-editor.js:2112:27)
  at Selection.module.exports.Selection.screenRangeChanged (C:\Users\jgreeley\AppData\Local\atom\app-1.0.7\resources\app.asar\src\selection.js:911:26)
  at C:\Users\jgreeley\AppData\Local\atom\app-1.0.7\resources\app.asar\src\selection.js:43:24
  at Emitter.module.exports.Emitter.emit (C:\Users\jgreeley\AppData\Local\atom\app-1.0.7\resources\app.asar\node_modules\event-kit\lib\emitter.js:82:11)
  at Marker.module.exports.Marker.notifyObservers (C:\Users\jgreeley\AppData\Local\atom\app-1.0.7\resources\app.asar\src\marker.js:270:27)
  at C:\Users\jgreeley\AppData\Local\atom\app-1.0.7\resources\app.asar\src\marker.js:65:26
  at Emitter.module.exports.Emitter.emit (C:\Users\jgreeley\AppData\Local\atom\app-1.0.7\resources\app.asar\node_modules\event-kit\lib\emitter.js:82:11)
  at Marker.module.exports.Marker.emitChangeEvent (C:\Users\jgreeley\AppData\Local\atom\app-1.0.7\resources\app.asar\node_modules\text-buffer\lib\marker.js:401:20)
  at Marker.module.exports.Marker.update (C:\Users\jgreeley\AppData\Local\atom\app-1.0.7\resources\app.asar\node_modules\text-buffer\lib\marker.js:346:12)
  at Marker.module.exports.Marker.setRange (C:\Users\jgreeley\AppData\Local\atom\app-1.0.7\resources\app.asar\node_modules\text-buffer\lib\marker.js:108:19)
  at Marker.module.exports.Marker.setBufferRange (C:\Users\jgreeley\AppData\Local\atom\app-1.0.7\resources\app.asar\src\marker.js:137:32)
  at C:\Users\jgreeley\AppData\Local\atom\app-1.0.7\resources\app.asar\src\selection.js:117:24
  at Selection.module.exports.Selection.modifySelection (C:\Users\jgreeley\AppData\Local\atom\app-1.0.7\resources\app.asar\src\selection.js:940:7)
  at Selection.module.exports.Selection.setBufferRange (C:\Users\jgreeley\AppData\Local\atom\app-1.0.7\resources\app.asar\src\selection.js:110:19)
  at Selection.module.exports.Selection.selectWord (C:\Users\jgreeley\AppData\Local\atom\app-1.0.7\resources\app.asar\src\selection.js:404:12)
  at TextEditorComponent.module.exports.TextEditorComponent.onMouseDown (C:\Users\jgreeley\AppData\Local\atom\app-1.0.7\resources\app.asar\src\text-editor-component.js:556:42)
  at HTMLDivElement.<anonymous> (C:\Users\jgreeley\AppData\Local\atom\app-1.0.7\resources\app.asar\src\text-editor-component.js:3:61)

```

### Commands

```
     -3:31.5.0 emmet:balance-outward (atom-text-editor.editor.is-focused)
     -3:23.8.0 window:toggle-menu-bar (atom-text-editor.editor.is-focused)
  2x -3:21.4.0 emmet:balance-outward (atom-text-editor.editor.is-focused)
     -3:04.9.0 window:toggle-menu-bar (div.settings-view.pane-item)
     -1:30.2.0 settings-view:check-for-package-updates (atom-workspace.workspace.scrollbars-visible-always.theme-atom-material-syntax.theme-atom-material-ui)
     -0:17.1.0 find-and-replace:show (atom-text-editor.editor.is-focused)
  7x -0:15.7.0 core:confirm (atom-text-editor.editor.mini.is-focused)
     -0:05.3.0 editor:consolidate-selections (atom-text-editor.editor.is-focused)
     -0:05.3.0 core:cancel (atom-text-editor.editor.is-focused)
```

### Config

```json
{
  "core": {
    "themes": [
      "atom-material-ui",
      "atom-material-syntax"
    ],
    "ignoredNames": [
      ".git",
      ".hg",
      ".svn",
      ".DS_Store",
      "._*",
      "Thumbs.db"
    ],
    "autoHideMenuBar": true
  }
}
```

### Installed Packages

```coffee
# User
atom-material-syntax, v0.2.3
atom-material-ui, v0.5.21
atom-perforce, v1.5.3
color-picker, v2.0.11
dartlang, v0.3.16
emmet, v2.3.12
file-icons, v1.6.5
go-plus, v3.4.5
language-batch, v0.3.3
linter, v1.4.2
minimap, v4.12.3
pigments, v0.11.0

# Dev
No dev packages
```

**Comment by Alhadis** [comment]
Got this too. Was AFK for two minutes to confirm something with one of the accounts managers, came back, double-clicked, BAM! Suddenly I'm seeing this.

![psigeoa](https://cloud.githubusercontent.com/assets/2346707/9434027/6257edb8-4a7d-11e5-962f-b008879f571e.jpg)

**Comment by fungilation** [comment]
Same. Happened after pressing cmd-e and cmd-g to search for instances of a highlighted string.

Atom QA (or lack thereof) is getting shitty. The modified indicators on tabs are still broken for multiple versions now.

**Comment by benogle** [comment]
> The modified indicators on tabs are still broken for multiple versions now.

This is fixed in 1.0.19.

**Comment by fungilation** [comment]
Seems like it. But it did something worse through the last few updates: reverted code tabs to older code versions! (though marking those tabs as modified while at it)

**Comment by benogle** [comment]
> reverted code tabs to older code versions!

I'm not sure I understand. Can you make a new issue outlining the problem?

**Comment by 50Wliu** [comment]
> reverted code tabs to older code versions! (though marking those tabs as modified while at it)

I believe this was also fixed in 1.0.19.

**Comment by fungilation** [comment]
So far so good I think
